### PR TITLE
Closes CPPL-220

### PR DESCRIPTION
**Pull request by iodream**
Added a single GoBack button at the top level, removed GoBack buttons on sub pages

### CODE REVIEW

**Review by MaksymTerekhovskyy — APPROVED** [review]

File: src/Client/Controller/MainPage/NavHistory.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@ void NavHistory::Clear()
 
 MainSubPages NavHistory::GetLastPage()
 {
+	if(m_storage.size() == 0)
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/MainPage/NavHistory.cpp:19]: Stack has method `empty` for this.

File: src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#include "IncomeEditSubPage.h"
+#include "ui_IncomeEditSubPage.h"
+
+IncomeEditSubPage::IncomeEditSubPage(QWidget *parent) :
+	QWidget(parent),
+	ui(new Ui::IncomeEditSubPage)
+{
+	ui->setupUi(this);
+
+	ui->ExpirationDateEdit->setDate(QDate::currentDate());
```
Comment (yaroslav-kernitskyi) [inline src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp:10]: I'm not sure ui should do this

**Review by yaroslav-kernitskyi — APPROVED** [review]
I didn't found any major issues. Also i think it would be better to put refactoring of "go back" button and implementing ui components in separate branches

**Comment by iodream** [comment]
> I didn't found any major issues. Also i think it would be better to put refactoring of "go back" button and implementing ui components in separate branches

It was in separate branches but i thought i'd merge with Maksym's branch first to make sure it works, which was not merged at that time yet. Hence some of the issues which are in fact part of that other pull request, which is now merged.

File: src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp (hunk 2 above)
Comment (iodream) [reply src/Client/View/MainPage/Income/IncomeEditSubPage/IncomeEditSubPage.cpp:10]: i will take another look at it when i implement the update functions on the controller

**Pull request merged**